src/Microsoft.AspNet.Mvc.TagHelpers/ScriptTagHelper.cs

```diff
@@ -51,83 +125,168 @@ public class ScriptTagHelper : TagHelper
+            if (fallbackSrcs.Any())
             {
-                if (!attribute.Key.Equals(SrcAttributeName, StringComparison.OrdinalIgnoreCase))
-                {
-                    var encodedKey = JavaScriptStringEncoder.Default.JavaScriptStringEncode(attribute.Key);
-                    var encodedValue = JavaScriptStringEncoder.Default.JavaScriptStringEncode(attribute.Value);
+                // Build the <script> tag that checks the test method and if it fails, renders the extra script.
+                builder.AppendLine();
+                builder.Append("<script>(")
+                   .Append(FallbackTestExpression)
+                   .Append("||");
 
-                    content.AppendFormat(CultureInfo.InvariantCulture, " {0}=\\\"{1}\\\"", encodedKey, encodedValue);
-                }
-                else
+                builder.Append("document.write(\"");
+
+                foreach (var src in fallbackSrcs)
                 {
-                    AppendSrc(content, attribute.Key);
+                    builder.Append("<script");
+
+                    if (!attributes.ContainsKey("src"))
+                    {
+                        AppendSrc(builder, "src", src);
+                    }
+
+                    foreach (var attribute in attributes)
+                    {
+                        if (!attribute.Key.Equals(SrcAttributeName, StringComparison.OrdinalIgnoreCase))
+                        {
+                            var encodedKey = JavaScriptStringEncoder.Default.JavaScriptStringEncode(attribute.Key);
+                            var encodedValue = JavaScriptStringEncoder.Default.JavaScriptStringEncode(attribute.Value);
+
+                            builder.AppendFormat(
+                                CultureInfo.InvariantCulture,
+                                " {0}=\\\"{1}\\\"",
+                                encodedKey,
+                                encodedValue);
+                        }
+                        else
+                        {
+                            AppendSrc(builder, attribute.Key, src);
+                        }
+                    }
+
+                    builder.Append("><\\/script>");
                 }
+                
+                builder.Append("\"));</script>");
             }
+        }
 
-            content.Append("><\\/script>\"));</script>");
+        private void EnsureGlobbingUrlBuilder()
+        {
+            if (GlobbingUrlBuilder == null)
+            {
+                GlobbingUrlBuilder = new GlobbingUrlBuilder(
+                    HostingEnvironment.WebRootFileProvider,
+                    Cache,
+                    ViewContext.HttpContext.Request.PathBase);
+            }
+        }
+
+        private static void BuildScriptTag(
+            string content,
+            IDictionary<string, string> attributes,
+            StringBuilder builder)
+        {
+            builder.Append("<script");
```

Comment:
I took a look at `TagBuilder` when I did `LinkTagHelper` and decided it didn't actually suit the needs, i.e. I'd have to change it to use it effectively. Can't remember specifics but it might've been to do with it encoding attributes you add, and in this case most of the attributes being added are already encoded.